packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -55,7 +58,7 @@ export abstract class BaseFetchService<
   protected abstract preLoopHook(data: {valid: boolean; startHeight: number}): Promise<void>;
 
   constructor(
-    protected apiService: ApiService,
+    protected apiService: ApiService<A, SA, BC>,
```

Comment:
You can set defaults on the ApiService generics so we don't need to provide generics everywhere. This is useful here were fetch.service doesn't really care about the generic types of api service. The same can be applied in other places.

Alternatively if you want to narrow down the type `BaseFetchService` coulld have `API extends ApiService` rather than all the individual generics of `ApiService`